### PR TITLE
Updated for commons-configuration2 in PinotConfiguartion

### PR DESCRIPTION
**Pull request by abhioncbr**
As per the [issue](https://github.com/apache/pinot/issues/11085) and in continuation of the previous work https://github.com/apache/pinot/pull/11792 & https://github.com/apache/pinot/pull/11868,  This PR upgrade the `PinotConfiguartion` to `commons-configuartion2`.

cc: @xiangfu0 / @Jackie-Jiang / @walterddr

### CODE REVIEW

File: pinot-spi/src/main/java/org/apache/pinot/spi/env/CommonsConfigurationUtils.java [hunk 1]
```diff
@@ -32,27 +32,54 @@
...
 public class CommonsConfigurationUtils {
+  private static final Character DEFAULT_LIST_DELIMITER = ',';
   private CommonsConfigurationUtils() {
   }
 
+  public static void setDefaultListDelimiterHandler(PropertiesConfiguration configuration) {
+    configuration.setListDelimiterHandler(new DefaultListDelimiterHandler(DEFAULT_LIST_DELIMITER));
+  }
+
+  public static void loadPropertiesConfiguration(PropertiesConfiguration configuration, String path)
```
Comment (Jackie-Jiang) [inline pinot-spi/src/main/java/org/apache/pinot/spi/env/CommonsConfigurationUtils.java:55]: Same for other loading methods, we can directly return `PropertiesConfiguration`
```suggestion
  public static PropertiesConfiguration loadFromFile(String filePath)
```

File: pinot-core/src/test/java/org/apache/pinot/core/query/executor/QueryExecutorExceptionsTest.java [hunk 2]
```diff
@@ -155,8 +156,8 @@ public void setUp()
     resourceUrl = getClass().getClassLoader().getResource(QUERY_EXECUTOR_CONFIG_PATH);
     assertNotNull(resourceUrl);
     PropertiesConfiguration queryExecutorConfig = new PropertiesConfiguration();
-    queryExecutorConfig.setDelimiterParsingDisabled(false);
-    queryExecutorConfig.load(new File(resourceUrl.getFile()));
+    FileHandler fileHandler = new FileHandler(queryExecutorConfig);
```
Comment (Jackie-Jiang) [inline pinot-core/src/test/java/org/apache/pinot/core/query/executor/QueryExecutorExceptionsTest.java:159]: Use `CommonsConfigurationUtils.loadFromFile(resourceUrl.getFile())`, same for other places where config needs to be loaded

**Review by Jackie-Jiang — COMMENTED** [review]
Mostly good

**Comment by abhioncbr** [comment]
@Jackie-Jiang, Can you please review it once more? Thanks

**Review by Jackie-Jiang — APPROVED** [review]

**Pull request merged**